### ISSUE TITLE
Update `CvGame::loadPirateShip` to consider size matters for transporting.

### ISSUE THREAD

**Issue by billw2012** [caveman2cosmos/Caveman2Cosmos#278]
Ideally we should be entirely encapsulating all transport logic so that separate code paths are not required.

**Comment by Thunderbrd** [comment]
I completely agree and have been chewing on that concept in my head for a while now.  The logic to load for war is a little different, potentially a lot different, and logic for loading and unloading for land units is currently non-existent and the purpose differs a lot.  But at least for civilian and spying and hunting purposes and stuff related to a unit needing to get from area a to area b, it really should be much more generic-capable than it currently is.
